platform/default/mbgl/storage/offline_database.cpp

```diff
@@ -236,6 +245,20 @@ optional<std::pair<Response, uint64_t>> OfflineDatabase::getResource(const Resou
     return std::make_pair(response, size);
 }
 
+optional<int64_t> OfflineDatabase::hasResource(const Resource& resource) {
```

Comment:
This method should update `resources.accessed`, as `getResource` does. You can keep it to a single query by checking `db->changes()` after the update -- if it's 0, you know the resource does not exist.